packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1149,6 +1157,84 @@ export class EthModule {
     return this._node.uninstallFilter(filterId, true);
   }
 
+  // eth_feeHistory
+
+  private _feeHistoryParams(
+    params: any[]
+  ): [BN, RpcNewBlockTag, number[] | undefined] {
+    const [blockCount, newestBlock, rewardPercentiles] = validateParams(
+      params,
+      rpcQuantity,
+      rpcNewBlockTag,
+      optional(t.array(rpcFloat))
+    );
+
+    if (blockCount.ltn(1)) {
+      throw new InvalidInputError(`blockCount should be at least 1`);
+    }
+
+    if (blockCount.gtn(1024)) {
+      throw new InvalidInputError(`blockCount should be at most 1024`);
+    }
+
+    if (rewardPercentiles !== undefined) {
+      for (const [i, p] of rewardPercentiles.entries()) {
+        if (p < 0 || p > 100) {
+          throw new InvalidInputError(
+            `The reward percentile number ${
+              i + 1
+            } is invalid. It must be a float between 0 and 100, but is ${p} instead.`
+          );
+        }
+
+        if (i !== 0) {
+          const prev = rewardPercentiles[i - 1];
+          if (prev >= p) {
+            throw new InvalidInputError(
+              `The reward percentiles should be in increasing order, but the percentile number ${
+                i + 1
+              } is not larger than the previous one`
+            );
+          }
+        }
+      }
+    }
+
+    return [blockCount, newestBlock, rewardPercentiles];
+  }
+
+  private async _feeHistoryAction(
+    blockCount: BN,
+    newestBlock: RpcNewBlockTag,
+    rewardPercentiles?: number[]
+  ) {
+    if (!this._node.isEip1559Active()) {
+      throw new InvalidInputError(
+        `eth_feeHistory is disabled. It only works with the London hardfork or a later one.`
+      );
+    }
+
+    const resolvedNewestBlock = await this._resolveNewBlockTag(newestBlock);
+
+    const feeHistory = await this._node.getFeeHistory(
+      blockCount,
+      resolvedNewestBlock,
+      rewardPercentiles ?? []
```

Comment:
Nonce of the RPC methods handles their default params in their `_SomethingParams` method, so I think it's better to keep it in the action, for consistency.

Reply:
Oh, didn't notice that. Maybe at some point it would make sense to refactor them, but that's not probably something we'll do anytime soon, so I won't bother even opening an issue for this.